core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1289,7 +1289,9 @@ public void delete(AlluxioURI path, DeleteOptions options) throws IOException,
         auditContext.setAllowed(false);
         throw e;
       }
-      mMountTable.checkUnderWritableMountPoint(path);
+      if (!options.isAlluxioOnly()) {
+        mMountTable.checkUnderWritableMountPoint(path);
```

Comment:
When we create a file, we check whether it's under a writable mount point even if the write type is `MUST_CACHE`. The readonly nature of a mount point applies both to UFS space as well as Alluxio space. I don't think it's correct to skip this check just because `isAlluxioOnly` is set.

Reply:
@aaudiber In this `delete` method, if we want to delete alluxio metadata only, and if mount point is readonly, `mMountTable.checkUnderWritableMountPoint(path)` will throw a exception, the result is the delete operation can not be executed.

Reply:
Ok, I'm all right with this. @gpang what do you think of making it possible to `delete --alluxioOnly` from readonly mount points?